src/Adapter/Product/CommandHandler/UpdateProductStockHandler.php

```diff
@@ -0,0 +1,148 @@
+use Product;
+
+/**
+ * Handles @see UpdateProductStockCommand using legacy object model
+ */
+final class UpdateProductStockHandler extends AbstractProductHandler implements UpdateProductStockHandlerInterface
+{
+    /**
+     * @var ProductRepository
+     */
+    private $productRepository;
+
+    /**
+     * @var ProductStockUpdater
+     */
+    private $productStockUpdater;
+
+    /**
+     * @param ProductRepository $productRepository
+     * @param ProductStockUpdater $productStockUpdater
+     */
+    public function __construct(
+        ProductRepository $productRepository,
+        ProductStockUpdater $productStockUpdater
+    ) {
+        $this->productRepository = $productRepository;
+        $this->productStockUpdater = $productStockUpdater;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(UpdateProductStockCommand $command): void
+    {
+        $product = $this->productRepository->get($command->getProductId());
+
+        $propertiesToUpdate = $this->fillUpdatableProperties($product, $command);
+        $this->productStockUpdater->update($product, $propertiesToUpdate, $command->addMovement());
+    }
+
+    /**
+     * @param Product $product
+     * @param UpdateProductStockCommand $command
+     *
+     * @return array
+     */
+    private function fillUpdatableProperties(
+        Product $product,
+        UpdateProductStockCommand $command
+    ): array {
+        $updatableProperties = [];
+
+        if (null !== $command->getLocation()) {
+            $product->location = $command->getLocation();
+            $updatableProperties[] = 'location';
+        }
+        if (null !== $command->getLowStockAlert()) {
+            $product->low_stock_alert = $command->getLowStockAlert();
+            $updatableProperties[] = 'low_stock_alert';
+        }
+        if (null !== $command->getLowStockThreshold()) {
+            $product->low_stock_threshold = $command->getLowStockThreshold();
+            $updatableProperties[] = 'low_stock_threshold';
+        }
+        if (null !== $command->getMinimalQuantity()) {
+            $product->minimal_quantity = $command->getMinimalQuantity();
+            $updatableProperties[] = 'minimal_quantity';
+        }
+        if (null !== $command->getOutOfStockType()) {
+            $product->out_of_stock = $command->getOutOfStockType()->getValue();
+            $updatableProperties[] = 'out_of_stock';
+        }
+        if (null !== $command->getPackStockType()) {
+            $product->pack_stock_type = $command->getPackStockType()->getValue();
+            $updatableProperties[] = 'pack_stock_type';
+        }
+        if (null !== $command->getQuantity()) {
+            $product->quantity = $command->getQuantity();
+            $updatableProperties[] = 'quantity';
+        }
+        if (null !== $command->getAvailableDate()) {
+            $product->available_date = $command->getAvailableDate()->format('Y-m-d');
```

Comment:
in one of PR's (https://github.com/PrestaShop/PrestaShop/pull/21804/files#diff-9d738966e547d8db477408207db40533ec9cf8b178d4803c2952e91b318616dfR39)
I've added Util/DateTime::DEFAULT_FORMAT.
If it gets merged we could use it here

Reply:
Oh indeed nice, but it's not merged yet so I copied it into my PR

Reply:
I actually added two consts because we need one for DateTime and one for Date only

Reply:
I only added DEFAULT_DATE_FORMAT in the end to avoid conflicts and unused const, but I think you should rename yours DEFAULT_DATETIME_FORMAT in your PR to be consistent

Reply:
done :heavy_check_mark: